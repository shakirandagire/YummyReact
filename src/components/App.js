import React, { Component } from 'react';
import Notification from 'react-notify-toast';
import { BrowserRouter, Route, Switch } from 'react-router-dom';
import '../css/style.css';
import Main from './Main';
import Register from './Register';
import Login from './Login';
import AddCategory from './AddCategory';
import ViewCategory from './ViewCategory';
import EditCategory from './EditCategory';
import AddRecipe from './AddRecipe';
import ViewRecipe from './ViewRecipe';
import EditRecipe from './EditRecipe';
import PrivateRoute from './PrivateRoute';
import ResetPassword from './ResetPassword';

class App extends Component {
  render() {
    return (
      <BrowserRouter>
        <div>
          <Notification />
          <Switch>
            <Route exact path="/" component={Main} />
            <Route exact path="/register" component={Register} />
            <Route exact path="/login" component={Login} />
            <Route exact path="/reset-password" component={ResetPassword} />
            <PrivateRoute exact path="/add-categories" component={AddCategory} />
            <PrivateRoute exact path="/view-categories" component={ViewCategory} />
            <PrivateRoute exact path="/edit-categories/:id" component={EditCategory} />
            <PrivateRoute exact path="/add-recipes/:id/recipes" component={AddRecipe} />
            <PrivateRoute exact path="/view-recipes/:id/recipes" component={ViewRecipe} />
            <PrivateRoute exact path="/edit-recipes/:category_id/recipes/:id" component={EditRecipe} />
          </Switch>
        </div>
      </BrowserRouter>
    );
  }
}

export default App;
