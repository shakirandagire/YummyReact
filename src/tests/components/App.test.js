import React from 'react';
import App  from '../../components/App';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

const component = shallow(< App/>);

describe('<App/>',() => {
  it('should render properly', () =>{
    expect(component.length).toBe(1)
  })
  it('should render app',() =>{
    expect(shallowToJson(component)).toMatchSnapshot();
})
})

