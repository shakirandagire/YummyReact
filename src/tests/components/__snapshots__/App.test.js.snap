// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App/> should render app 1`] = `
<BrowserRouter>
  <div>
    <_class
      options={Object {}}
    />
    <Switch>
      <Route
        component={[Function]}
        exact={true}
        path="/"
      />
      <Route
        component={[Function]}
        exact={true}
        path="/register"
      />
      <Route
        component={[Function]}
        exact={true}
        path="/login"
      />
      <Route
        component={[Function]}
        exact={true}
        path="/reset-password"
      />
      <PrivateRoute
        component={[Function]}
        exact={true}
        path="/add-categories"
      />
      <PrivateRoute
        component={[Function]}
        exact={true}
        path="/view-categories"
      />
      <PrivateRoute
        component={[Function]}
        exact={true}
        path="/edit-categories/:id"
      />
      <PrivateRoute
        component={[Function]}
        exact={true}
        path="/add-recipes/:id/recipes"
      />
      <PrivateRoute
        component={[Function]}
        exact={true}
        path="/view-recipes/:id/recipes"
      />
      <PrivateRoute
        component={[Function]}
        exact={true}
        path="/edit-recipes/:category_id/recipes/:id"
      />
    </Switch>
  </div>
</BrowserRouter>
`;
